datasets/iwslt2017/iwslt2017.py

```diff
@@ -166,8 +166,8 @@ def _generate_examples(self, source_files, target_files, split):
         id_ = 0
         source, target = self.config.pair.split("-")
         for source_file, target_file in zip(source_files, target_files):
-            with open(source_file) as sf:
-                with open(target_file) as tf:
+            with open(source_file, "r", encoding="utf-8") as sf:
+                with open(target_file, "w", encoding="utf-8") as tf:
```

Comment:
```suggestion
                with open(target_file, "r", encoding="utf-8") as tf:
```

Reply:
Bitten by my own internal pattern matching (target files get written to)